homeassistant/components/xbox_live/sensor.py

```diff
@@ -140,7 +140,7 @@ def async_update(event_time=None):
 
     def update(self):
         """Update state data from Xbox API."""
-        presence = self._api.get_user_presence(self._xuid)
+        presence = self._api.gamer(gamertag="", xuid=self._xuid).get("presence")
```

Comment:
Can we guarantee that the `"presence"` key is in the result of this api call? If so we should not use `dict.get`. If we need to use `dict.get` to guard for a missing key, we also need to return a dict as default value if the key is missing in the dict. Otherwise there'll be an error below.

Reply:
The `get` does not actually refer to retrieving a value from a dict, it is the method for getting data from the API. `gamer()` instantiates the `Gamer` class, which then supports a few endpoints that you can reference with `get()` ([implementation](https://github.com/mKeRix/xboxapi-python/blob/master/xboxapi/gamer.py#L36)). That being said I updated the other `get` usages in the sensor to access the dict directly where possible.